R/knit_print.R

```diff
@@ -1,5 +1,21 @@
 knit_print.data.frame = function(x, ...) {
   context <- render_context()
+
+  # printing of certain expressions producing
+  # 'data.table' objects should be suppressed
+  printable <- TRUE
+  if ("data.table" %in% loadedNamespaces() &&
+      exists("shouldPrint", envir = asNamespace("data.table")))
+  {
+    printable <- tryCatch(
+      data.table:::shouldPrint(x) && inherits(x, "data.table"),
```

Comment:
FWIW I think the ordering of checks here might be important, since it might matter whether the promise `x` has been forced by the time it enters the `shouldPrint()` function.